sklearn/tree/_tree.pyx

```diff
@@ -197,6 +197,16 @@ cdef class Criterion:
 
         pass
 
+    cdef double approx_impurity_improvement(self) nogil:
+        """Approximate the impurity reduction removing constants for a split"""
```

Comment:
Could you please extend that docstring to make it explicit that this is the only part of the impurity improvement computation that is required to find the best split threshold for a given feature. This method is used to speed up the search for the best split.

The absolute impurity improvement is only computed  by the `impurity_improvement` method once the best split has been found.

Maybe calling that "approximate" is misleading (its a shifted and scaled version of the true value that preserves the location of the best split). But I don't personally have a better naming suggestion at hand.